crates/protocol/src/serde_envelope.rs

```diff
@@ -13,6 +13,17 @@ pub struct SerdeEnvelope<Contents> {
 }
 
 impl<Contents> SerdeEnvelope<Contents> {
+    /// Creates a new envelope for some contents using a signature.
+    ///
+    /// Note: this does not verify the signature matches the contents.
+    pub fn new(contents: Contents, key_id: signing::KeyID, signature: signing::Signature) -> Self {
```

Comment:
Is this needed somewhere? I would prefer if the only way to make these guaranteed signature consistency.

Reply:
We don't guarantee the signature on deserialization either; this is just a way of constructing an envelope from the constituent parts without using serialization.

Reply:
IMO, not guaranteeing on serialization/de-serialization is different because as long as you're not synthesizing serialized versions without using the serde it's still always going to be valid. This makes it possible to mess up in a more typical usage. As a compromise, can we make this `pub(crate)` or restricted in some other way?

Reply:
Agreed that having a `new` method vs. deserializing does make it more of a footgun.

It can't be `pub(crate)` as it's called from the server crate, but perhaps I can figure out a different approach.

Taking a step back, though, perhaps it's not valuable to have the individual columns on the `checkpoints` table and instead just persist the envelope itself as a JSON blob (like we do for validator state)?

Let me see what I can come up with.a

Reply:
I couldn't come up with something that wasn't horrible, and I do like that we have the columns in the `checkpoints` table rather than some serialized blob for the envelope, so I renamed the method to `from_parts_unchecked` to make it a little clearer that this method shouldn't be used to construct any ol' `SerdeEnvelope`.

Thoughts on a different approach?

Reply:
I think that's a good compromise. I like that `unchecked` makes it clear you can do this wrong if you aren't checking things yourself.